### DISCUSSION TITLE
Multiple Wyze Cams Clobbering Network

### DISCUSSION THREAD

**Discussion by fullhack2** [mrlt8/docker-wyze-bridge#135] [Q&A]
Hi all,

This isn't specifically a question/concern around the add-on, but I thought I may not be the only one in this boat.

I am running 4 V3's and 1 V2 currently.  When I start the add-on for all of them, it basically brings down my 2.4 network and other devices start dropping like flies.  It looks like the Wyze cams absolutely destroy the 2.4 channel utilization when streaming.

I am running an Asus AC86 router (Merlin)  with 2 Ubiquiti AC-LR AP's.  When the cameras aren't streaming and the add-on isn't running, channel utilization is minimal 10-15%, but as soon as I enable the add-on it jumps to 100% and starts killing things.

I have tried many combinations of AP tuning (power/rssi/channels) but to no avail.  I am running a dedicated 2.4 ssid for my IoT devices.

Any thoughts?  Anyone else running into this?

I have seen a few posts regarding Wyze cams and their horrible antennas as possibly the problem, but not sure what else can be done.

I was thinking is adding another AP to the mix to see if it would help, but I'm not sure it would really help.

Would love to use this as part of my NVR setup, but I can't reliably get the streams without bringing down everything else at the moment.

Help!

Thanks!

**Comment by lvdfrnt** [comment]
> Hi all,
> 
> This isn't specifically a question/concern around the add-on, but I thought I may not be the only one in this boat.
> 
> I am running 4 V3's and 1 V2 currently. When I start the add-on for all of them, it basically brings down my 2.4 network and other devices start dropping like flies. It looks like the Wyze cams absolutely destroy the 2.4 channel utilization when streaming.
> 
> I am running an Asus AC86 router (Merlin) with 2 Ubiquiti AC-LR AP's. When the cameras aren't streaming and the add-on isn't running, channel utilization is minimal 10-15%, but as soon as I enable the add-on it jumps to 100% and starts killing things.
> 
> I have tried many combinations of AP tuning (power/rssi/channels) but to no avail. I am running a dedicated 2.4 ssid for my IoT devices.
> 
> Any thoughts? Anyone else running into this?
> 
> I have seen a few posts regarding Wyze cams and their horrible antennas as possibly the problem, but not sure what else can be done.
> 
> I was thinking is adding another AP to the mix to see if it would help, but I'm not sure it would really help.
> 
> Would love to use this as part of my NVR setup, but I can't reliably get the streams without bringing down everything else at the moment.
> 
> Help!
> 
> Thanks!

I specifically logged on to see if anyone was having the same issue I am having and lo and behold, you're not alone! 

I have 5x V3's and 1x V2 and it is acting almost the same as you described. I'm a techy so my house is riddled with way too many IoT devices and I've never had issues with my network setup (3x Deco X20 + 2x Deco W3600). However ever since I stumbled upon this addon and installed it, my network has been acting up. IoT devices going offline, who network "freezing up" and having to reboot in order to get things back online. The cameras only last 2-3 hrs streaming through RTSP to my Amcrest NVR and then they either freeze the frame or drop connection to the RTSP feed. When that is happening, I can still stream through the Wyze app to view the cameras just fine. Usually, I can restart the add-on and cameras come back, but then only last 2-3 hrs and back in same loop.

I am not sure what is going on, I have tried same as you, dedicated SSID, network optimization through Deco app, prioritizing camera bandwidth and still nothing. 

I do notice that the port #s keep jumping in the log file but no stream when the cameras do no display through RTSP: 
```
2021/09/19 17:42:21 I [2/1] [RTSP] [conn 192.168.68.125:58401] opened
2021/09/19 17:42:21 I [2/1] [RTSP] [conn 192.168.68.125:58401] closed
2021/09/19 17:42:21 I [2/1] [RTSP] [conn 192.168.68.125:58400] closed
2021/09/19 17:42:21 I [2/1] [RTSP] [conn 192.168.68.125:58402] opened
2021/09/19 17:42:21 I [2/1] [RTSP] [conn 192.168.68.125:58403] opened
2021/09/19 17:42:21 I [2/1] [RTSP] [conn 192.168.68.125:58403] closed
2021/09/19 17:42:21 I [2/1] [RTSP] [conn 192.168.68.125:58404] opened
2021/09/19 17:42:21 I [2/1] [RTSP] [conn 192.168.68.125:58404] closed
2021/09/19 17:42:21 I [2/1] [RTSP] [conn 192.168.68.125:58405] opened
```

Long story short, I'm on the same boat as you and have not stumbled to find a solution...

**Comment by mrlt8** [comment]
Make sure you have airtime fairness off - wyze even has a [support page](https://support.wyze.com/hc/en-us/articles/360031479211-ASUS-Routers-and-Airtime-Fairness) about this

IMO, this is probably one of the reasons why wyze keeps refusing to add RTSP to the official firmware since you won't run into the issue until you start streaming from multiple cameras at the same time.

**Reply by fullhack2** [reply]
Hmmm...   I have now tried with both off and on.  Unfortunately I am still seeing the same results.  Thanks for the suggestion though.

**Comment by terryhonn** [comment]
Is there anything else besides disabling airtime fairness that might help this situation? I don't use an ASUS router and I see the same thing. After starting the bridge (2 v3, 3 v2 and a doorbell) within seconds multiple devices start falling off the network. If I shut down the bridge, everything comes back online. 
I'm not a network guru; would a separate VLAN for the cams do anything to help?

**Reply by mrlt8** [reply]
I don't think a vlan would help in the situation. Moving your 5Ghz devices off the 2.4Ghz band and disabling certain types of beam forming or other non-standard 'enhancements' might help since we have multiple cameras trying to send a constant stream of data at the same time. Might also want to play around with 20/40Mhz. One of my neighbor's wifi is on 40Mhz and can cause issues if they overlap with my 2.4Ghz channel.

You may be able to run something like this from a pi or other linux box to see if your router reports the channel utilization:
`sudo iw wlan0 scan | egrep 'SSID|primary channel|utilisation'`

**Comment by lvdfrnt** [comment]
I don’t have a ASUS either and have tried all optimizations with no luck.

Wyze just came out with their own RTSP firmware for v3 yesterday. I would say try that. I plan on trying it when I get home tomorrow. Saw a lot of success stories with people that had a lot of cams on same network.

**Reply by mrlt8** [reply]
Would be interesting to see if the official RTSP also suffers from the same wifi utilization issues!

**Reply by mikedepalma** [reply]
Did you ever find a resolution here? Running into same issue with 8 v2s.... in the process of setting up separate VLAN and SSID now to see if that works.

**Reply by mrlt8** [reply]
It's most likely a UDP issue, and because we're running each camera in a separate process, it tends to flood the network with even more UDP traffic. See #241  where @maxfield-allison has been looking into the cause of the issue.

**Reply by maxfield-allison** [reply]
Fyi Wyze has even taken down all of their RTSP firmware since its so old and they likely kept running into similar issues in testing.

**Reply by BeardedTek** [reply]
From what I've found, it's the router that is the issue here.
Most people use a "cheap" (not as in money) router that is meant to handle some congestion, but as soon as it gets ramped up with streaming video, udp packets, iot devices, etc it starts to buckle as it's CPU and memory overloads.

I currently have 3 Wyze V3 cameras, 2 1080p Merkury (tuya) cameras with rtsp enabled, several esp32-cams for motion detection/presence, and a webcam on a raspberry pi all streaming 24x7 through frigate and frigate Event Video Recorder.

I'm running a Cisco 800 series router with 2 TP-Link Routers in AP mode in a mesh configuration and a Linksys with WRT software running in AP mode as well.  Passing the routing onto something more substantial allows for the udp to just pass through the access points and be processed with something a bit meatier that can handle the load.

I'd humbly suggest getting a cheap older used mini pc with a couple NICs and a couple gigs of RAM, throw openWRT, ipfire, or pfsense on it, and Bob's your uncle, you're not so bottlenecked anymore.

**Reply by maxfield-allison** [reply]
I'm running 4 unifi AP's and 3 switches, none of which are encountering CPU or memory utilization issues. Everything feeds into a Intel(R) Core(TM) i5-4590 CPU @ 3.30GHz (4 cores, 4 threads) 16GB OPNsense firewall for routing. I doubt the issue, in my case at least, is incapable hardware. specifically I've see high channel utilization on the AP-AC-LR when multiple cams are connected. This doesn't occur when running the RTSP firmware but only when this container is active and those cams are connected to the same AP. #241 has much more detail.

**Comment by BeardedTek** [comment]
Hmmm...  Maybe I'll fire up good ol Wireshark and see what's happening on my network.  I currently am running wyze-bridge and the rtsp firmware on my V3's.  Maybe this can provide some insight.

William J Kenny III
The Bearded Tek
***@***.***
https://u22838684.ct.sendgrid.net/ls/click?upn=qENo-2BaQ-2F0-2BrOhG-2Fgfyebc-2FrV3ftULRgXpbvEwnoHuCLWBXdEBV4-2FTpgfVySLE-2FbPeuYz_U31jt5HgHBh7k3Grc5s4l4yAjU0ikYzFwdTzzW090Qc3RE-2FC-2BUpP2-2Ba-2FPUOgpSpjiwPIpzFbCGcEK7MfLUgbP-2Ftc27CJzz02LtlP7WaiN0k-2FRg9eBiOjrlLUvIKVS9P0QDVAZAqJyE1Q-2BbJhT9CvlBz-2Boa7JnqWYeDFKn7oH-2F5-2FXghDpQeVlaDFvQJLYfZnND0-2B97BJTuk-2B1EIe1r5DVJ9gB-2BJ7LrkgpNOt0W96X336Y7uYF-2Bxl10fajvXIgAPZegE89XfS6hITG-2FXSFkVNxHw-3D-3D
https://u22838684.ct.sendgrid.net/ls/click?upn=qENo-2BaQ-2F0-2BrOhG-2Fgfyebc-2F7l8k23p8vaRjsOtVXy4d-2F15YS5m9ZMraMapqk-2FvgkCksie_U31jt5HgHBh7k3Grc5s4l4yAjU0ikYzFwdTzzW090Qc3RE-2FC-2BUpP2-2Ba-2FPUOgpSpjiwPIpzFbCGcEK7MfLUgbP-2Ftc27CJzz02LtlP7WaiN0k-2FRg9eBiOjrlLUvIKVS9P0-2BXEgTBGDb45IwjZWO9f2QYszuKFOpfw-2BI4pqlNq26cRVXP1lMVoFEdKYrfZztSiPFIjQ2OvwweG4l-2FNsWFHXmzzhSnx4oZ7xTPbKWqm5mvYyKFC58m2TiEuF10I6cPvpo7E3Ly05XSyUNQJCYmqvuQ-3D-3D

**Comment by wirsmann** [comment]
Not sure if this helps but I'm using a Deco mesh network (3x Deco x60 2x Deco m9) with 6 v3 cameras and experience the same issue.  As soon as I run the docker container my network gets flooded - easy to tell, wyze app stops working well, google home can't communicate to IoT devices.  Stop the container and things go back to normal.

**Reply by maxfield-allison** [reply]
do they have wired backhaul or are they bridging? it would make sense for your network to get hammered even harder with the wireless mesh. I'm interested to hear its happening on tp link systems too.

**Reply by mrlt8** [reply]
Check your router settings - Multicast rate, IGMP, STP, Beamforming, etc.

**Reply by benchristian88** [reply]
We have exactly the same issue on Unifi.   igmp snooping on, rstp on, proxy arp on.  The usage just goes through the roof, like every camera is streaming the entire time. which actually might be the case.  the port stats show that with 20mb or so additional usage.  (i have 15 cameras).  About to try change the ondemand setting to true, and see what happens

**Reply by roger-** [reply]
Should IGMP, beam forming, etc. be disabled? Curious how/why are those leading to the issue.

**Reply by mrlt8** [reply]
IGMP snooping enabled to filter out unneeded wyze traffic flooding your entire network. Disabling beam forming may help as it may introduce additional issues with the camera's wifi.

**Comment by benchristian88** [comment]
I have updated ON_DEMAND=True and that seems to have solved it for me. My 2.4 is back where is should be, and my streaming is still there, and snapshots work perfectly via scrypted into Homekit. I dont fully understand the different between continuous streaming, and how that will impact things such as motion detection or recording events, as I havent played with those features yet, but at least the wifi frequency is not being hammered now.

**Reply by mrlt8** [reply]
I believe the pre buffer and motion detection options in scrypted require a constant connection, so that may keep an active connection to multiple cameras.
